File: javascript/src/arrays/0027-remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
const removeElement = function (nums, val) {
  let k = 0;

  for (let r = 0; r < nums.length; r++) {
    if (nums[r] !== val) {
      nums[k] = nums[r];
      k++;
    }
  }

  return k;
};

module.exports = removeElement;

File: golang/arrays/0027-remove-element.go
package arrays

func removeElement(nums []int, val int) int {
	if len(nums) == 0 {
		return 0
	}

	k := 0

	for r := 0; r < len(nums); r++ {
		if nums[r] != val {
			nums[k] = nums[r]
			k++
		}
	}

	return k
}
